lib/Core/BuildEngine.cpp

```diff
@@ -631,6 +644,13 @@ class BuildEngineImpl {
     while (true) {
       bool didWork = false;
 
+      // Cancel the build, if requested.
+      if (buildCancelled) {
+        // Force completion of all outstanding tasks.
+        cancelRemainingTasks();
+        return false;
```

Comment:
I'm not sure... this typically happens because the delegate asked us to cancel (I think we still auto-cancel if no delegate is registered in some cases, but probably makes sense to eventually get rid of that).

WDYT?

Reply:
Yeah, most likely not necessary.  It occurred to me based on how the test works.  I was thinking that the delegate might want to know if something else cancelled the build.  But in the 'real world' such calls would most likely come from the delegate, especially if we refactor this into a build context of in the future.